test/plugin/test_out_detect_exceptions.rb

```diff
@@ -98,6 +116,45 @@ def test_exception_detection
     assert_equal(make_logs(t, *messages), d.events)
   end
 
+  def test_ignore_nested_exceptions
+    test_cases = {
+      "php": PHP_EXC,
+      "python": PYTHON_EXC,
+      "ruby": RUBY_EXC
+    }
+
+    test_cases.each do |language, exception|
+      cfg = "languages #{language}"
+      d = create_driver(cfg)
+      t = Time.now.to_i
+
+      # Convert exception to a single line to simplify the test case.
+      single_line_exception = exception.gsub("\n", "\\n")
+
+      # There is a nested exception within the body, we should ignore those!
+      json_line_with_exception = %({"timestamp": {"nanos": 998152494, "seconds": 1496420064}, "message": "#{single_line_exception}", "thread": 139658267147048, "severity": "ERROR"}\n)
+      json_line_without_exception = %({"timestamp": {"nanos": 5990266, "seconds": 1496420065}, "message": "next line", "thread": 139658267147048, "severity": "INFO"}\n)
+
+      router_mock = flexmock('router')
+
+      # Validate that each line received is emitted separately as expected.
+      router_mock.should_receive(:emit).once.with(DEFAULT_TAG, Integer, {
+        "message" => json_line_with_exception, 
+        "count" => 0
+      })
+      router_mock.should_receive(:emit).once.with(DEFAULT_TAG, Integer, {
+        "message" => json_line_without_exception, 
+        "count" => 1
+      })
+
+      d.instance.router = router_mock
+
+      d.run do
+        feed_lines(d, t, json_line_with_exception + json_line_without_exception)
+      end
+    end
```

Comment:
The chance of this might be low. Yet what if the router receives additional messages except these two? I'd recommend still asserting / checking `d.events` at the end.